### ISSUE TITLE
Wrong C++ runtime source distribution file permission & .idea folder

### ISSUE THREAD

**Issue by Xeonacid** [antlr/antlr4#4507]
I downloaded C++ runtime source distribution from https://www.antlr.org/download/antlr4-cpp-runtime-4.13.1-source.zip , it seems many source file permissions are 755 instead of 644, and it contains a `.idea` folder.
The `.idea` folder will be installed as an empty folder when doing `make install`.
![image](https://github.com/antlr/antlr4/assets/13995937/a33ac3cc-606e-44f5-8c4e-c574cad28e84)

Please consider fixing the file permission and removing `.idea` folder. Thanks!

**Comment by kaby76** [comment]
You are correct. https://github.com/antlr/antlr4/blob/ebb511a04a60ae5a605aba65471c07dd854e9303/runtime/Cpp/runtime/src/ANTLRFileStream.cpp displays "Executable File" in the header just before the numbered listing of the source code. So, the file was checked in wrong. On Ubuntu 20.04, unzip extracts the files with that "x" permission.
